src/InstagramClone/InstagramClone.Web/Controllers/ImageController.cs

```diff
@@ -0,0 +1,38 @@
+﻿using InstagramClone.Web.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace InstagramClone.Web.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class ImageController : ControllerBase
+    {
+        private static List<Image> _images = new List<Image>();
+
+        [HttpPost("upload")]
+        public async Task<IActionResult> Upload([FromForm] string title, [FromForm] string uploader, [FromForm] string description, [FromForm] IFormFile file)
+        {
+            var newImage = new Image()
+            {
+                Title = title,
+                Description = description,
+                Uploader = uploader,
+                UploadDate = DateTime.Now,
+            };
+
+            using (var stream = new MemoryStream())
+            {
+                await file.CopyToAsync(stream);
+                newImage.ImageData = stream.ToArray();
+            }
+
+            _images.Add(newImage);
+            return Ok();
```

Comment:
Kérték, hogy minél specifikusabban adjunk vissza értéket a frontendre szerintem érdemes lenne, már az elejétől figyelni erre és a kivételekre/hibákra is gondolni hogy akkor mit adjunk vissza. De ha azt mondjátok, hogy az majd meg lesz később, akkor jó most így is.

Reply:
@palatinb Csináltam DTO-t ahogyan kérted. Nem tudom mire gondolsz a kivételek/hibák alatt.
"Web API controllers don't have to check ModelState.IsValid if they have the [ApiController] attribute. In that case, an automatic HTTP 400 response containing error details is returned when model state is invalid. For more information, see [Automatic HTTP 400 responses.](https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-5.0#automatic-http-400-responses)"

Reply:
igazából arra gondoltam, hogyha bármilyen futási idejű hiba lenne akkor ne mindig 400-al térjünk vissza, hanem mondjuk valamilyen specifikusabb HTTP válasszal, de nyilván a default lehet a 400

Reply:
Ja persze, de azt majd később, amikor már ténylegesen el lesz tárolva valahova, gondolom entity framework segítségével.
Szerintem az első demo-hoz elég lesz ennyi.

Reply:
okés, bocsi nem fogalmaztam tisztán 😄